### ISSUE TITLE
`lua require'lspconfig'.rust_analyzer.setup({})` results in segfault ~ 30% of the time

### ISSUE THREAD

**Issue by Aaronik** [rust-lang/rust-analyzer#14050]
**rust-analyzer version**: (eg. output of "rust-analyzer: Show RA Version" command, accessible in VSCode via <kbd>Ctrl/⌘</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd>)

I'm on neovim and using mason, and :Mason shows me that the version is `installed version 2023-01-23`, although I can't find a semver version anywhere

EDIT: Digging around deep in the bowels of my machine's subfolders I found the bin and ran it with --version and got `rust-analyzer 0.3.1377-standalone` 

**rustc version**: (eg. output of `rustc -V`)

Freshly updated: `rustc 1.67.0 (fc594f156 2023-01-24)`

**relevant settings**: (eg. client settings, or environment variables like `CARGO`, `RUSTC`, `RUSTUP_HOME` or `CARGO_HOME`)

All of the aforementioned env vars are empty.

I'm using neovim, and:

```
nvim --version

NVIM v0.9.0-dev-613+gef18c9f9b
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -DNVIM_TS_HAS_SET_ALLOCATOR -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/build/nvim/parts/nvim/build/build/cmake.config -I/build/nvim/parts/nvim/build/src -I/build/nvim/parts/nvim/build/.deps/usr/include -I/usr/include -I/build/nvim/parts/nvim/build/build/src/nvim/auto -I/build/nvim/parts/nvim/build/build/include
Compiled by root@lcy02-amd64-005

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

:checkhealth gives:

```

==============================================================================
lazy: require("lazy.health").check()

lazy.nvim ~
- OK Git installed
- OK no existing packages found by other package managers
- OK packer_compiled.lua not found
- ERROR Issues were reported when loading your specs:
- WARNING {mason.nvim}: setting a table to `Plugin.config` is deprecated. Please use `Plugin.opts` instead
- WARNING {mason-lspconfig.nvim}: setting a table to `Plugin.config` is deprecated. Please use `Plugin.opts` instead

==============================================================================
mason: require("mason.health").check()

mason.nvim report ~
- OK neovim version >= 0.7.0
- WARNING **Go**: not available
- OK **cargo**: `cargo 1.67.0 (8ecd4f20a 2023-01-10)`
- WARNING **luarocks**: not available
- OK **Ruby**: `ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-linux-gnu]`
- OK **RubyGem**: `3.3.5`
- WARNING **Composer**: not available
- WARNING **PHP**: not available
- OK **npm**: `8.19.2`
- OK **node**: `v18.12.1`
- OK **python3**: `Python 3.10.6`
- WARNING **pip3**: not available
- WARNING **javac**: not available
- OK **java**: `openjdk version "11.0.17" 2022-10-18`
- WARNING **julia**: not available
- OK **wget**: `GNU Wget 1.21.2 built on linux-gnu.`
- OK **curl**: `curl 7.81.0 (x86_64-pc-linux-gnu) libcurl/7.81.0 OpenSSL/3.0.2 zlib/1.2.11 brotli/1.0.9 zstd/1.4.8 libidn2/2.3.2 libpsl/0.21.0 (+libidn2/2.3.2) libssh/0.9.6/openssl/zlib nghttp2/1.43.0 librtmp/2.3 OpenLDAP/2.5.13`
- OK **gzip**: `gzip 1.10`
- OK **tar**: `tar (GNU tar) 1.34`
- WARNING **pwsh**: not available
- OK **bash**: `GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)`
- OK **sh**: `Ok`
- OK GitHub API rate limit. Used: 1. Remaining: 59. Limit: 60. Reset: Sun 29 Jan 2023 04:55:01 PM PST.

==============================================================================
nvim: require("nvim.health").check()

Configuration ~
- OK no issues found

Runtime ~
- OK $VIMRUNTIME: /snap/nvim/2791/usr/share/nvim/runtime

Performance ~
- OK Build type: RelWithDebInfo

Remote Plugins ~
- OK Up to date

terminal ~
- key_backspace (kbs) terminfo entry: `key_backspace=\177`
- key_dc (kdch1) terminfo entry: `key_dc=\E[3~`
- $VTE_VERSION="6800"
- $COLORTERM="truecolor"

==============================================================================
provider: health#provider#check

Clipboard (optional) ~
- OK Clipboard tool found: xclip

Python 3 provider (optional) ~
- `g:python3_host_prog` is not set.  Searching for python3 in the environment.
- Multiple python3 executables found.  Set `g:python3_host_prog` to avoid surprises.
- Executable: /usr/bin/python3
- Other python executable: /bin/python3
- Python version: 3.10.6
- pynvim version: 0.4.2 (outdated; from /usr/lib/python3/dist-packages/neovim)
- WARNING Latest pynvim is NOT installed: 0.4.3

Python virtualenv ~
- OK no $VIRTUAL_ENV

Ruby provider (optional) ~
- Ruby: ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-linux-gnu]
- WARNING `neovim-ruby-host` not found.
  - ADVICE:
    - Run `gem install neovim` to ensure the neovim RubyGem is installed.
    - Run `gem environment` to ensure the gem bin directory is in $PATH.
    - If you are using rvm/rbenv/chruby, try "rehashing".
    - See :help |g:ruby_host_prog| for non-standard gem installations.
    - You may disable this provider (and warning) by adding `let g:loaded_ruby_provider = 0` to your init.vim

Node.js provider (optional) ~
- Node.js: v18.12.1
- WARNING Missing "neovim" npm (or yarn, pnpm) package.
  - ADVICE:
    - Run in shell: npm install -g neovim
    - Run in shell (if you use yarn): yarn global add neovim
    - Run in shell (if you use pnpm): pnpm install -g neovim
    - You may disable this provider (and warning) by adding `let g:loaded_node_provider = 0` to your init.vim

Perl provider (optional) ~
- WARNING "Neovim::Ext" cpan module is not installed
  - ADVICE:
    - See :help |provider-perl| for more information.
    - You may disable this provider (and warning) by adding `let g:loaded_perl_provider = 0` to your init.vim

==============================================================================
vim.lsp: require("vim.lsp.health").check()

- LSP log level : WARN
- Log path: /home/aaron/.local/state/nvim/lsp.log
- Log size: 17008 KB

==============================================================================
vim.treesitter: require("vim.treesitter.health").check()

- Nvim runtime ABI version: 14
- OK Parser: ada        ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/ada.so
- OK Parser: agda       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/agda.so
- OK Parser: arduino    ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/arduino.so
- OK Parser: astro      ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/astro.so
- OK Parser: awk        ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/awk.so
- OK Parser: bash       ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/bash.so
- OK Parser: beancount  ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/beancount.so
- OK Parser: bibtex     ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/bibtex.so
- OK Parser: blueprint  ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/blueprint.so
- OK Parser: c          ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/c.so
- OK Parser: c_sharp    ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/c_sharp.so
- OK Parser: clojure    ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/clojure.so
- OK Parser: cmake      ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/cmake.so
- OK Parser: comment    ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/comment.so
- OK Parser: commonlisp ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/commonlisp.so
- OK Parser: cooklang   ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/cooklang.so
- OK Parser: cpp        ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/cpp.so
- OK Parser: css        ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/css.so
- OK Parser: cuda       ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/cuda.so
- OK Parser: dart       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/dart.so
- OK Parser: diff       ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/diff.so
- OK Parser: dockerfile ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/dockerfile.so
- OK Parser: dot        ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/dot.so
- OK Parser: ebnf       ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/ebnf.so
- OK Parser: eex        ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/eex.so
- OK Parser: elixir     ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/elixir.so
- OK Parser: elm        ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/elm.so
- OK Parser: elvish     ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/elvish.so
- OK Parser: embedded_template ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/embedded_template.so
- OK Parser: erlang     ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/erlang.so
- OK Parser: fennel     ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/fennel.so
- OK Parser: fish       ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/fish.so
- OK Parser: foam       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/foam.so
- OK Parser: fortran    ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/fortran.so
- OK Parser: fusion     ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/fusion.so
- OK Parser: gdscript   ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/gdscript.so
- OK Parser: git_rebase ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/git_rebase.so
- OK Parser: gitattributes ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/gitattributes.so
- OK Parser: gitcommit  ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/gitcommit.so
- OK Parser: gleam      ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/gleam.so
- OK Parser: glimmer    ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/glimmer.so
- OK Parser: glsl       ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/glsl.so
- OK Parser: go         ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/go.so
- OK Parser: gomod      ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/gomod.so
- OK Parser: gowork     ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/gowork.so
- OK Parser: graphql    ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/graphql.so
- OK Parser: hack       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/hack.so
- OK Parser: haskell    ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/haskell.so
- OK Parser: hcl        ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/hcl.so
- OK Parser: heex       ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/heex.so
- OK Parser: help       ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/help.so
- OK Parser: hjson      ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/hjson.so
- OK Parser: hlsl       ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/hlsl.so
- OK Parser: hocon      ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/hocon.so
- OK Parser: html       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/html.so
- OK Parser: http       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/http.so
- OK Parser: java       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/java.so
- OK Parser: javascript ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/javascript.so
- OK Parser: jq         ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/jq.so
- OK Parser: jsdoc      ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/jsdoc.so
- OK Parser: json       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/json.so
- OK Parser: json5      ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/json5.so
- OK Parser: jsonc      ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/jsonc.so
- OK Parser: jsonnet    ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/jsonnet.so
- OK Parser: julia      ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/julia.so
- OK Parser: kotlin     ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/kotlin.so
- OK Parser: lalrpop    ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/lalrpop.so
- OK Parser: latex      ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/latex.so
- OK Parser: ledger     ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/ledger.so
- OK Parser: llvm       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/llvm.so
- OK Parser: lua        ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/lua.so
- OK Parser: m68k       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/m68k.so
- OK Parser: make       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/make.so
- OK Parser: markdown   ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/markdown.so
- OK Parser: markdown_inline ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/markdown_inline.so
- OK Parser: menhir     ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/menhir.so
- OK Parser: mermaid    ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/mermaid.so
- OK Parser: meson      ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/meson.so
- OK Parser: nickel     ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/nickel.so
- OK Parser: ninja      ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/ninja.so
- OK Parser: nix        ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/nix.so
- OK Parser: norg       ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/norg.so
- OK Parser: ocaml      ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/ocaml.so
- OK Parser: ocaml_interface ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/ocaml_interface.so
- OK Parser: org        ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/org.so
- OK Parser: pascal     ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/pascal.so
- OK Parser: perl       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/perl.so
- OK Parser: php        ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/php.so
- OK Parser: phpdoc     ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/phpdoc.so
- OK Parser: pioasm     ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/pioasm.so
- OK Parser: prisma     ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/prisma.so
- OK Parser: proto      ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/proto.so
- OK Parser: pug        ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/pug.so
- OK Parser: python     ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/python.so
- OK Parser: ql         ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/ql.so
- OK Parser: qmljs      ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/qmljs.so
- OK Parser: query      ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/query.so
- OK Parser: r          ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/r.so
- OK Parser: racket     ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/racket.so
- OK Parser: rasi       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/rasi.so
- OK Parser: regex      ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/regex.so
- OK Parser: rego       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/rego.so
- OK Parser: rnoweb     ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/rnoweb.so
- OK Parser: rst        ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/rst.so
- OK Parser: ruby       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/ruby.so
- OK Parser: rust       ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/rust.so
- OK Parser: scala      ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/scala.so
- OK Parser: scheme     ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/scheme.so
- OK Parser: scss       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/scss.so
- OK Parser: slint      ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/slint.so
- OK Parser: solidity   ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/solidity.so
- OK Parser: sparql     ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/sparql.so
- OK Parser: supercollider ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/supercollider.so
- OK Parser: surface    ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/surface.so
- OK Parser: svelte     ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/svelte.so
- OK Parser: sxhkdrc    ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/sxhkdrc.so
- OK Parser: t32        ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/t32.so
- OK Parser: terraform  ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/terraform.so
- OK Parser: tiger      ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/tiger.so
- OK Parser: tlaplus    ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/tlaplus.so
- OK Parser: todotxt    ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/todotxt.so
- OK Parser: toml       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/toml.so
- OK Parser: tsx        ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/tsx.so
- OK Parser: turtle     ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/turtle.so
- OK Parser: twig       ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/twig.so
- OK Parser: typescript ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/typescript.so
- OK Parser: v          ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/v.so
- OK Parser: vala       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/vala.so
- OK Parser: verilog    ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/verilog.so
- OK Parser: vhs        ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/vhs.so
- OK Parser: vim        ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/vim.so
- OK Parser: vue        ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/vue.so
- OK Parser: wgsl       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/wgsl.so
- OK Parser: wgsl_bevy  ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/wgsl_bevy.so
- OK Parser: yaml       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/yaml.so
- OK Parser: yang       ABI: 13, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/yang.so
- OK Parser: zig        ABI: 14, path: /snap/nvim/2791/usr/share/nvim/runtime/parser/zig.so
```

Following a thread in that checkhealth command, looking in `~/.local/state/nvim/lsp.log`, I'm seeing this error, among other errors from other LSP servers:

`[ERROR][2022-12-13 19:07:44] .../vim/lsp/rpc.lua:733  "rpc" "rust-analyzer" "stderr"  '[ERROR rust_analyzer::dispatch] unknown request: Request { id: RequestId(I32(344)), method: "workspace/              
     executeCommand", params: Object({"title": String("triggerParameterHints"), "command": String("editor.action.triggerParameterHints")}) }\n'` -- I don't know if this is related to the segfaults.

Here is my complete neovim config in lua, with an empty .vimrc:

```lua
-- https://github.com/folke/lazy.nvim
local lazypath = vim.fn.stdpath("data") .. "/lazy/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "--single-branch",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)

require("lazy").setup({
  {
    'williamboman/mason.nvim',
    config = {}
  },

  {
    'williamboman/mason-lspconfig.nvim',
    config = {
      ensure_installed = { 'rust_analyzer' }
    },
    init = function()
      require('mason-lspconfig').setup_handlers {
        ["rust_analyzer"] = function()
          require("lspconfig")["rust_analyzer"].setup({})
        end,

      }
    end
  },

  {
    'neovim/nvim-lspconfig',
  },
})

```

I know this is heavily neovim related but I'm sincerely helping y'all can help me figure this out. It only happens when I run `:lua require("lspconfig")["rust_analyzer"].setup {}`, whether that happens in my config or if I comment it out in the config and then run it manually after startup.

This issue has been a major thorn in my side since I started working with Rust a few months back. Today I spent nearly all day trying to figure it out. At first I had no idea what was causing it, and today I narrowed it down to this LSP server.

It's possible it's `lspconfig` specific... I guess I'll wait for a reply here and then make a ticket there to the same effect.

**Comment by Veykril** [comment]
rust-analyzer doesn't support the `workspace/executeCommand` request, it also does not set a capability for it that it would support it. So this seems like an implementation error on the client side.

**Comment by flodiebold** [comment]
What exactly segfaults though? An unsupported request shouldn't make either RA or neovim segfaults.

**Comment by Veykril** [comment]
Iirc we panic on unhandled requests

**Comment by Aaronik** [comment]
Dang there's info from another log that I forgot to include.

I'm on Ubuntu:

```
$ uname -a
Linux dellbert 5.15.0-58-generic #64-Ubuntu SMP Thu Jan 5 11:43:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

and in `/var/log/syslog` is how I know it's a `segfault`:

```
Jan 29 11:30:40 dellbert kernel: [521432.461742] nvim[245283]: segfault at 7fe1737a9ba2 ip 0000561a3b032117 sp 00007ffc01bd6078 error 4 in nvim[561a3ad4f000+349000]
```

Also neovim exits with code `139` for whatever that's worth.

It sounds like this ball might not be in y'all's court, but I figure it can't hurt for you to be aware that this is happening at least to one random dude :P

**Comment by lnicola** [comment]
If it's `nvim` that segfaults, that's not our code, right?

**Comment by Aaronik** [comment]
Interesting yeah you're right, nvim should not segfault no matter what your code does, even if yours is the only one I can find that this happens with. Maybe it is the implementation. I guess i can follow up with nvim-lsp, since they seem to be making some calls that aren't supported,and with nvim itself, which shouldn't be crashing no matter what you or nvim-lsp is doing

**Comment by Aaronik** [comment]
Ok just to update: I made [a ticket](https://github.com/neovim/nvim-lspconfig/issues/2431) in `nvim-lspconfig`, which I believe is what's responsible for sending the unsupported request. I'm not sure if y'all really care that much, but in case you do, now you know. Unfortunately that ticket was cursory closed and I don't believe it was actually read. But at least I tried.

I don't think there's much more for me that can be done from here so I'll close this issue. Thank you guys for working with me on this!
